### DISCUSSION TITLE
Coming Soon: Actual versioning

### DISCUSSION THREAD

**Discussion by ikeycode** [AerynOS/discussions#9] [Announcements]
Hi all! We've been mulling lately on how we should version stuff, on account of the ISO version scheme is erratic and randomly picked, and all of our tooling is stuff at `0.1.0` ala Cargo.toml defaults.

We've opted for the following format:

    $STONE_VERSION.YY.NUM
    
Where `$STONE_VERSION` is the current internal stone format version (previously known as `moss format`) shared by `moss`, `boulder` and the infrastructure. This is presently `0` until we're satisfied fluidity has settled. `YY` will obviously be the current year, 2 digit format.

`NUM` is a simple, incrementing integer, to reduce the cognitive load for versioning, and reduce pressure to "justify" tagging a new release. This will also decouple stability promises from tooling versions, eliminating the need for milestones-to-1.0.

For example, a version might be `0.24.1` to indicate the unstable format, in `2024`, and the bump number being `1`.

This will soon be reflected in moss internals, and will pave the way for *versioned repositories*.

**TLDR**

Right now your install probably says `prealpha0` or `0.5`. Soon it will be `0.something-that-makes-sense`.

**Comment by ikeycode** [comment]
This has now been implemented in `moss` and `boulder` in the unified `tools` repository:

https://github.com/serpent-os/tools/commit/06780d277d5bc703239114902cda55b4a9ff7bae

**Comment by ikeycode** [comment]
![image](https://github.com/user-attachments/assets/944177ff-d689-4c6a-a184-8881e81aa2c8)
